packages/app/src/app/overmind/effects/vscode/index.ts

```diff
@@ -528,139 +571,143 @@ export class VSCodeEffect {
   }
 
   private async loadEditor(monaco: any, container: HTMLElement) {
-    this.monaco = monaco;
-    this.workbench = new Workbench(monaco, this.controller, this.runCommand);
+    return new Promise(async vscodeResolved => {
```

Comment:
Oh yes, I wrapped `loadEditor` in a promise and only let it resolve when the editor layout is mounted. I'm not sure if this is better?